### PR TITLE
lambda.part2.exercise2 (Seliukov)

### PR DESCRIPTION
**Pull request by Duelist256**

### CODE REVIEW

File: src/test/java/lambda/part2/exercise/Exercise2.java [hunk 1]
```diff
@@ -13,66 +13,61 @@
...
     }
 
     @Test
     public void personHasNotEmptyLastNameAndFirstNameUsingLogicalOperators() {
-        Predicate<Person> personHasEmptyFirstName = null;
-        Predicate<Person> personHasEmptyLastName = null;
+        Predicate<Person> personHasEmptyFirstName =
+                negateUsingLogicalOperator(person -> !person.getFirstName().isEmpty());
+        Predicate<Person> personHasEmptyLastName =
+                negateUsingLogicalOperator(person -> !person.getLastName().isEmpty());
```
Comment (a-filipp0v) [inline src/test/java/lambda/part2/exercise/Exercise2.java]: Double negations? why?

File: src/test/java/lambda/part2/exercise/Exercise2.java [hunk 2]
```diff
@@ -13,66 +13,61 @@
...
 
-        Predicate<Person> personHasNotEmptyFirstName = null;
-        Predicate<Person> personHasNotEmptyLastName = null;
+        Predicate<Person> personHasNotEmptyFirstName =
+                negateUsingLogicalOperator(person -> person.getFirstName().isEmpty());
+        Predicate<Person> personHasNotEmptyLastName =
+                negateUsingLogicalOperator(person -> person.getLastName().isEmpty());
```
Comment (a-filipp0v) [inline src/test/java/lambda/part2/exercise/Exercise2.java]: You can just negate personHasEmpty*Name.

File: src/test/java/lambda/part2/exercise/Exercise2.java [hunk 3]
```diff
@@ -13,66 +13,61 @@
...
 
-        Predicate<Person> personHasNotEmptyLastNameAndFirstName = null;
+        Predicate<Person> personHasNotEmptyLastNameAndFirstName =
+                andUsingLogicalOperator(person -> !person.getFirstName().isEmpty(),
+                        person -> !person.getLastName().isEmpty());
```
Comment (a-filipp0v) [inline src/test/java/lambda/part2/exercise/Exercise2.java]: Use previous predicates as an arguments.

**Comment by elefus** [comment]
@a-filipp0v, подтверди изменения, если все замечания исправлены

File: src/test/java/lambda/part2/exercise/Exercise2.java [hunk 4]
```diff
@@ -13,66 +13,60 @@
...
     }
 
     @Test
     public void personHasNotEmptyLastNameAndFirstNameUsingLogicalOperators() {
-        Predicate<Person> personHasEmptyFirstName = null;
-        Predicate<Person> personHasEmptyLastName = null;
+        Predicate<Person> personHasEmptyFirstName =
+                negateUsingLogicalOperator(person -> person.getFirstName().isEmpty());
+        Predicate<Person> personHasEmptyLastName =
+                negateUsingLogicalOperator(person -> person.getLastName().isEmpty());
```
Comment (a-filipp0v) [inline src/test/java/lambda/part2/exercise/Exercise2.java]: Look carefully at variables names and ur assignment.

File: src/test/java/lambda/part2/exercise/Exercise2.java [hunk 5]
```diff
@@ -13,66 +13,60 @@
...
 
-        Predicate<Person> personHasNotEmptyFirstName = null;
-        Predicate<Person> personHasNotEmptyLastName = null;
+        Predicate<Person> personHasNotEmptyFirstName =
+                negateUsingLogicalOperator(negateUsingLogicalOperator(personHasEmptyFirstName));
+        Predicate<Person> personHasNotEmptyLastName =
+                negateUsingLogicalOperator(negateUsingLogicalOperator(personHasEmptyLastName));
```
Comment (a-filipp0v) [inline src/test/java/lambda/part2/exercise/Exercise2.java]: Why double negations again?

**Pull request merged**